tool/inits/GADGET_hdf5_PAR_IC.py

```diff
@@ -0,0 +1,118 @@
+
+            # output attribute name
+            output.write(format("%s:"%Header_key_i,"<25"))
+
+            # output attribute value(s) as string(s)
+            if isinstance(Header_list.attrs[Header_key_i],(list,np.ndarray)):
+                str_place_holder ='['+",  ".join(map(str, Header_list.attrs[Header_key_i]))+"]"
+            else: str_place_holder = str(Header_list.attrs[Header_key_i])
+
+            output.write(str_place_holder)
+            output.write("\n")
+
+        output.close()
+
+    print('Input_Parameters.txt complete')
+
+
+    # start sorting the particle attributes and initial conditions
+    # (1) ordinary point masses should have non-zero entry in the "MassTable"
+    GADGET_MassTable = list(Header_list.attrs["MassTable"])
+    # (2) "NumPart_ThisFile" should be identical to "NumPart_Total" IFF "input_GADGET_file" is NOT split over several smaller files
+    GADGET_NumPart_ThisFile = list(Header_list.attrs["NumPart_ThisFile"])
+    # (3) record the number of particles for each type
+    GADGET_NumPart_Total = list(Header_list.attrs["NumPart_Total"])
+    GADGET_NumPart_Total_HighWord = list(Header_list.attrs["NumPart_Total_HighWord"])
+    GADGET_BoxSize = Header_list.attrs["BoxSize"]
+
+    with open('PAR_IC', 'wb') as output:
+
+        for i, NumPart_i in enumerate(GADGET_NumPart_Total):
+            #print(i,NumPart_i, type(NumPart_i))
+            # if for "PartType(i)" the total particle number is greater than zero, we import the information to PAR_IC
+            if NumPart_i > 0:
+                key_NOW = "PartType"+str(i)
+                PartType_Coordinates = np.array(f[key_NOW]["Coordinates"], float)
+                #ParticleIDs_Coordinates = np.array(f[key_NOW]["ParticleIDs"], int)
+                Velocities_Coordinates = np.array(f[key_NOW]["Velocities"], float)
+
+                # see "UnitMass in g" on p.26 of https://wwwmpa.mpa-garching.mpg.de/gadget/users-guide.pdf
+                mass = np.full(NumPart_i, GADGET_MassTable[i])*mass_conversion
+
+                # see "BoxSize" on p.18 of https://wwwmpa.mpa-garching.mpg.de/gadget/users-guide.pdf
+                # set the origin of Cartesian coordinates to the box left bottom corner
+                posx = PartType_Coordinates[:,0]*pol_conversion
+                posy = PartType_Coordinates[:,1]*pol_conversion
+                posz = PartType_Coordinates[:,2]*pol_conversion
+
+                # see "Particle velocities" on p.32 of "https://wwwmpa.mpa-garching.mpg.de/gadget/users-guide.pdf"
+                velx = Velocities_Coordinates[:,0]*vel_conversion*(input_a_scale_factor_sqrt**3)
+                vely = Velocities_Coordinates[:,1]*vel_conversion*(input_a_scale_factor_sqrt**3)
+                velz = Velocities_Coordinates[:,2]*vel_conversion*(input_a_scale_factor_sqrt**3)
+
+                # see "Table 3" on p.31 of https://wwwmpa.mpa-garching.mpg.de/gadget/users-guide.pdf
+                partype = np.full(NumPart_i, i)
```

Comment:
The definitions of particle type in gamer and gadget are different. So, for now, we should fix the gamer particle type to `1` representing [PTYPE_GENERIC_MASSIVE](https://github.com/gamer-project/gamer/blob/master/include/Macro.h#L514).